express-zod-api/src/io-schema.ts

```diff
@@ -1,77 +1,64 @@
+/** @desc The type allowed on the top level of Middlewares and Endpoints */
+export type IOSchema = z.ZodType<Base>;
 
 /**
  * @description intersects input schemas of middlewares and the endpoint
  * @since 07.03.2022 former combineEndpointAndMiddlewareInputSchemas()
  * @since 05.03.2023 is immutable to metadata
  * @since 26.05.2024 uses the regular ZodIntersection
- * @see copyMeta
+ * @see mixExamples
  */
 export const getFinalEndpointInputSchema = <
-  MIN extends IOSchema<"strip">,
+  MIN extends IOSchema,
   IN extends IOSchema,
 >(
   middlewares: AbstractMiddleware[],
   input: IN,
 ): z.ZodIntersection<MIN, IN> => {
   const allSchemas: IOSchema[] = R.pluck("schema", middlewares);
   allSchemas.push(input);
-  const finalSchema = allSchemas.reduce((acc, schema) => acc.and(schema));
+  const finalSchema = allSchemas.reduce((acc, schema) =>
+    z.intersection(acc, schema),
+  );
   return allSchemas.reduce(
-    (acc, schema) => copyMeta(schema, acc),
+    (acc, schema) => mixExamples(schema, acc),
     finalSchema,
   ) as z.ZodIntersection<MIN, IN>;
 };
 
-export const extractObjectSchema = (
-  subject: IOSchema,
-): z.ZodObject<z.ZodRawShape> => {
+export const extractObjectSchema = (subject: IOSchema): z.ZodObject => {
   if (subject instanceof z.ZodObject) return subject;
-  if (subject instanceof z.ZodBranded)
-    return extractObjectSchema(subject.unwrap());
+  if (subject instanceof z.ZodInterface) {
+    const { optional } = subject._zod.def;
+    const mask = R.zipObj(optional, Array(optional.length).fill(true));
+    const partial = subject.pick(mask);
+    const required = subject.omit(mask);
+    return z
+      .object(required._zod.def.shape)
+      .extend(z.object(partial._zod.def.shape).partial());
+  }
   if (
     subject instanceof z.ZodUnion ||
     subject instanceof z.ZodDiscriminatedUnion
   ) {
-    return subject.options
-      .map((option) => extractObjectSchema(option))
-      .reduce((acc, option) => acc.merge(option.partial()), z.object({}));
-  } else if (subject instanceof z.ZodEffects) {
-    return extractObjectSchema(subject._def.schema);
-  } else if (subject instanceof z.ZodPipeline) {
-    return extractObjectSchema(subject._def.in);
-  } // intersection left:
-  return extractObjectSchema(subject._def.left).merge(
-    extractObjectSchema(subject._def.right),
-  );
+    return subject._zod.def.options
+      .map((option) => extractObjectSchema(option as IOSchema))
+      .reduce((acc, option) => acc.extend(option.partial()), z.object({}));
+  }
+  if (subject instanceof z.ZodPipe)
+    return extractObjectSchema(subject.in as IOSchema);
+  if (subject instanceof z.ZodIntersection) {
+    return extractObjectSchema(subject._zod.def.left as IOSchema).extend(
+      extractObjectSchema(subject._zod.def.right as IOSchema),
+    );
+  }
+  throw new IOSchemaError("Can not flatten IOSchema", { cause: subject });
```

Comment:
the method is used by `Diagnostics`. This should be handled somehow, or it should return same fallback?

Reply:
fixed by 86c56652